### PR TITLE
[exporter/dynatraceexporter] Update serialization 

### PR DESCRIPTION
**Pull request by pirgeo**
**Description:** This PR updates how certain OpenTelemetry internal representations are serialized to the Dynatrace wire format. Also, some deprecated fields were updated. Serialization methods were renamed to be more speaking. Overall, this PR adds a few things but is mostly cleanup and maintenance. 

**Testing:** Added tests to make sure data is serialized correctly.

### CODE REVIEW

File: exporter/dynatraceexporter/internal/serialization/sum.go [hunk 1]
```diff
@@ -39,6 +41,73 @@ func serializeSum(name, prefix string, dims dimensions.NormalizedDimensionList,
...
+			"dropping delta non-monotonic sum",
+			zap.String("name", metric.Name()),
+		)
+		return metricLines
+	}
+
+	points := metric.Sum().DataPoints()
+	numPoints := points.Len()
+
+	for i := 0; i < numPoints; i++ {
```
Comment (dyladan) [inline exporter/dynatraceexporter/internal/serialization/sum.go:59]:
```suggestion
	points := sum.DataPoints()
	for i := 0; i < points.Len(); i++ {
```

File: exporter/dynatraceexporter/internal/serialization/sum.go [hunk 2]
```diff
@@ -39,6 +41,73 @@ func serializeSum(name, prefix string, dims dimensions.NormalizedDimensionList,
 	return "", nil
 }
 
+func serializeSum(logger *zap.Logger, prefix string, metric pmetric.Metric, defaultDimensions dimensions.NormalizedDimensionList, staticDimensions dimensions.NormalizedDimensionList, prev *ttlmap.TTLMap, metricLines []string) []string {
+	sum := metric.Sum()
+	monotonic := sum.IsMonotonic()
+
+	if !monotonic && sum.AggregationTemporality() == pmetric.MetricAggregationTemporalityDelta {
```
Comment (dyladan) [inline exporter/dynatraceexporter/internal/serialization/sum.go:48]: monotonic var only used in one place, no need to create a var for it. `monotonic` doesn't convey any more info than `sum.IsMonotonic()`

```suggestion

	if !sum.IsMonotonic() && sum.AggregationTemporality() == pmetric.MetricAggregationTemporalityDelta {
```

**Review by dyladan — APPROVED** [review]
Couple very minor nits. This has already been reviewed internally at Dynatrace

File: exporter/dynatraceexporter/internal/serialization/sum.go (hunk 2 above)
Comment (pirgeo) [reply exporter/dynatraceexporter/internal/serialization/sum.go]: It's used in the loop down below as well. I updated it, especially in the loop below it reads more clearly that way I think.

File: CHANGELOG.md [hunk 3]
```diff
@@ -30,6 +30,7 @@
 - `transformprocessor`: Add IsMatch factory function.  This function allows regex matching in conditions (#10903)
 - `transformprocessor`: `replace_pattern` and `replace_all_patterns` use regex for pattern matching and replacing text in attributes/metrics. (#11125)
 - `coralogixexporter`: Add support for metrics (#11065)
+- `dynatraceexporter`: Update serialization (#11828)
```
[the remark this answers is not in the document]
Comment (dyladan) [reply CHANGELOG.md]: yeah looks good

**Review by codeboten — APPROVED** [review]

**Pull request merged**